### ISSUE TITLE
Gap in timeline

### ISSUE THREAD

**Issue by sandzwerg** [tuskyapp/Tusky#2518]
Sometimes, usually when I haven't checked tusky for a while and thus have a load of timeline to "load more" I get a gap in the timeline.

Usually this happens only if there is a big timeframe between "when I last checked tusky" and now, in the order of multiple hours but it seems to correlate with the activity in my timeline. I'm able to load a lot of toots but then the "load more" button is gone even when I'm pretty sure there are more posts that a not yet visible to me but I neither have a "load more" button nor a error that I indicate any kind of rate limit. IIRC the Mastodon servers have a API rate-limit so it might be that I'm hitting that but I'm not sure.
For example see this screenshoot where I have a "gap" of ~3h during the busy evening where I'm sure I'm missing posts (for example because I later see replies to a post I missed that was sent during this time)
![Tusky-timeline-gap](https://user-images.githubusercontent.com/272385/167941285-6b149021-e9cc-43ae-a691-faff6a74bee7.jpg)

I'm not 100% able to reproduce it but usually it happens when:

- Read tusky to the top of your TL
- don't open tusky for multiple hours
- open tusky, load the new toots from the bottom (via scrolling down after pressing the "load more" button)
- after a while there might be said gap in the TL

* * * *
- Tusky Version: 17.0 (Google)
- Android Version: 11
- Android Device: OnePlus Nord CE 5g
- Mastodon instance (if applicable): chaos.social

This might be related to the already closed issues #1022 and #399

**Comment by charlag** [comment]
Hi
Not ruling out that something is broken on our side but also sounds like you run into normal Mastodon behavior: Mastodon only keeps latest 400 posts of home timeline so if you start somewhere in the middle and scroll to the bottom you likely reach it very fast (you say it's related to activity so it sounds like it)

**Comment by sandzwerg** [comment]
yes that's certainly possible but I think in the past I got some kind of error. Is it possible to get more than the last 400 posts or is that a hard limit on server side?

**Comment by charlag** [comment]
There's no error, it's just a sliding window of 400 posts.
https://github.com/mastodon/mastodon/blob/bda8345e91b0e1d7b2c0534d09cb53973cb9ab18/app/lib/feed_manager.rb#L10

**Comment by sandzwerg** [comment]
Alright, thanks for clearing that up. Would it be possible to show a notice that one has reached that limit?

**Comment by charlag** [comment]
It might be possible with some effort. I'm not even sure if it's okay to show older posts as they might be already changed or deleted.

**Comment by nikclayton** [comment]
@charlag I'm not sure this is correct. The "400 posts in the home timeline" reason would only be the case if the user had > 400 new posts to read since their last refresh. It's possible that's what's happening here, but that would be a lot of new posts in the ~3h that @sandzwerg mentions.

**Comment by sandzwerg** [comment]
Considering that I follow around 850 accounts that is not completely out of the question.
